src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,507 @@ title: Builder.io & Astro
 description: Add content to your Astro project using Builder.io’s visual CMS
 type: cms
 service: Builder.io
-stub: true
+stub: false
 i18nReady: false
 ---
 import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import FileTree from '~/components/FileTree.astro';
 
+[Builder.io](https://www.builder.io/) is a visual CMS that supports drag and drop content editing for building web sites.
 
+At the end of this recipe you'll be able to create blog posts in the Builder app and have your Astro site update with this new content automatically using a webhook.
+
+To do this, you'll use Builder's [Content API](https://www.builder.io/c/docs/content-api) and [Qwik API](https://www.builder.io/c/docs/qwik-api) to connect your Builder space to Astro with zero client-side JavaScript.
+
+:::tip[Official resource]
+If you are using a JavaScript framework (e.g. Svelte, Vue, or React) in your Astro project you can use one of Builder's integrations as an alternative to making raw fetch calls through the REST API.
+:::
+
+## Prerequisites
+
+To get started, you will need to have the following:
+
+* **A Builder account and space** - If you don't have an account yet, [sign up for free](https://www.builder.io/) and create a new space. If you already have a space with Builder, feel free to use it, but you will need to modify the code to match the model name and custom data fields.
+* **A Builder API key** - This public key will be used to fetch your content from Builder. [Read Builder's guide on how your key](https://www.builder.io/c/docs/using-your-api-key#finding-your-public-api-key).
+
+## Setting up credentials
+
+To add your Builder API key to Astro, create a `.env` file in the root of your project with the following variable:
+
+```ini title=".env"
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+```
+
+Now, you should be able to use this API key in your project.
+
+:::note
+At the time of writing, [this key is public](https://www.builder.io/c/docs/using-your-api-key#prerequisites), so you don't have to worry about hiding or encrypting it.
+:::
```

Comment:
```suggestion
```
suggest remove.

Reply:
So, I'll point out that the key is named "PUBLIC_KEY"  :smile:  I think we could just adjust the sentence before to add the word public:

> Now, you should be able to use this public API key in your project.

As you can see, all our call outs are visually HUGE, and there's good reason for that - they need to draw attention to important info that we're afraid people might miss. We do try to keep them to a minimum and only use them when really necessary. I think given that this key has public in its name, and we can add one more public in the text, this note really shouldn't be necessary.

Reply:
I see your point. I also see a difference between using the word 'public' in a variable name or a description vs. stating it like 'this key is ok to share' :)

Just reading 'public API key' I can see myself thinking both 'ah, it's public as in it's ok to share' and 'ah, they're saying it's gonna be public, soooo.... I have to hide it?'

An explicit explanation eliminates this ambiguity. 

But I get the cost of more text and drawing a line of beginner-friendliness somewhere. I personally generally feel the need for docs to be more explicit way more than the need for them to be less verbose. That's why I'm pushing back here.